main/com.spotify/index.js

```diff
@@ -710,54 +724,48 @@ module.exports = class SpotifyDevice extends Tp.BaseDevice {
         return album;
     }
 
-    _findCommonArtist(songs, env) {
-        var artists = [];
-        const songObjs = this._queryResults.get(env.app.uniqueId);
-        if (!songObjs)
-            return null;
-        for (const song of songs) {
-            const result = songObjs[String(song)];
-            if (!result) {
-                return null;
-            } else {
-                if (artists.length === 0) {
-                    artists = result.artists.map((artist) => String(artist));
-                } else {
-                    const songArtists = result.artists.map((artist) => String(artist));
-                    artists = artists.filter((value) => songArtists.includes(value));
-                    if (artists.length === 0)
-                        return null;
-                }
-            }
-        }
-        if (artists.length !== 1)
-            return null;
-        else
-            return artists[0];
-    }
-
     async do_player_pause() {
         if (this._testMode())
             return;
+        const player_info = await this.get_get_play_info();
+        if (player_info) {
+            if (player_info.actions.disallows.pausing)
+                throw new Error("Failed to pause");
```

Comment:
We need to throw an error with the correct `code` property (like we do for `no_active_device`) and then add the error code and message to the `#_[on_error]` annotation for each function. And we should use a more descriptive error message explaining what is going on (tell people that they need to press play on the player or something).